tuf/api/metadata.py

```diff
@@ -1384,8 +1510,33 @@ def _is_target_in_pathpattern(targetpath: str, pathpattern: str) -> bool:
 
         return True
 
+    def find_delegation(self, target_filepath: str) -> Optional[str]:
+        """Find the delegated rolename based on the given ``target filepath``.
+
+        If the given ``target_filepath`` is not in any of the paths that
+        ``DelegatedRole`` is trusted to provide it returns None.
+
+        Args:
+            target_filepath: URL path to a target file, relative to a base
+                targets URL.
+        """
+        # Check if the given target_filepath is in any of the "paths" or
+        # "path_hash_prefixes" that DelegatedRole is trusted to provide.
+        warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
It's still not this simple. This filters all deprecation warnings within the application from this point forwards, this is incorrect for two reasons:
* we want to filter only warnings from this module
* we want to filter for only a single call, not all calls from now on

You would need to manually remove the filter, or better yet use the catchwarnings() context manager. 

This complexity is why I'm not necessarily convinced the deprecation warnings are worth it: they are surprisingly difficult to get right.

Reply:
Okay, for now, I will keep it simple and leave a docstring warning.
If we later decide to add a deprecation warning it could be worth it to explore, but probably for a more critical API than an API from a helper class.